testing/run-scripts/throughput.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python3
+
+import argparse
+import csv
+import io
+import subprocess
+import sys
+import time
+import urllib.parse
+
+FLOOD_SIZE_MB = 64
+FLOOD_MAX_SPEED = '5M'
+
+parser = argparse.ArgumentParser(
+    description='Measure SOCKS proxy throughput across browser versions.')
+parser.add_argument('clone_path', help='path to pre-built uproxy-lib repo')
+args = parser.parse_args()
+
+# Where is flood server?
+flood_ip = subprocess.check_output('./flood.sh ' + str(FLOOD_SIZE_MB) + ' ' +
+    FLOOD_MAX_SPEED, shell=True, universal_newlines=True).strip()
+print('** using flood server at ' + str(flood_ip))
+
+browsers = ['chrome', 'firefox']
+versions = ['stable', 'beta', 'canary']
+
+# Run the benchmarks.
+throughput = {}
+for browser in browsers:
+  for version in versions:
+    print('** ' + browser + '/' + version)
+
+    result = 0
+    try:
+      # Start the relevant config, stopping the previous one if necessary.
+      # TODO: check first if running, to avoid spurious warnings
+      subprocess.call(['docker', 'rm', '-f', 'uproxy-getter', 'uproxy-giver'])
+      spec = browser + '-' + version
+      subprocess.call('./run_pair.sh -p ' + args.clone_path + ' ' +
```

Comment:
Good catch. I think I tried that and it didn't work for `flood.sh` because I had a dumb typo in the opening line (`#/bin/bash` vs. `#!/bin/bash`).  All warning signs point against `shell=True`, so I'm using the `[]` form throughout.